src/ReadonlyArray.ts

```diff
@@ -0,0 +1,1678 @@
+/**
+ * @since 2.5.0
+ */
+import { Alternative1 } from './Alternative'
+import { Applicative } from './Applicative'
+import { Compactable1, Separated } from './Compactable'
+import { Either } from './Either'
+import { Eq } from './Eq'
+import { Extend1 } from './Extend'
+import { FilterableWithIndex1 } from './FilterableWithIndex'
+import { Foldable1 } from './Foldable'
+import { FoldableWithIndex1 } from './FoldableWithIndex'
+import { Predicate, Refinement, identity } from './function'
+import { FunctorWithIndex1 } from './FunctorWithIndex'
+import { HKT } from './HKT'
+import { Monad1 } from './Monad'
+import { Monoid } from './Monoid'
+import { ReadonlyNonEmptyArray } from './ReadonlyNonEmptyArray'
+import { isSome, none, Option, some } from './Option'
+import { fromCompare, getMonoid as getOrdMonoid, Ord, ordNumber } from './Ord'
+import { pipeable } from './pipeable'
+import { Show } from './Show'
+import { TraversableWithIndex1 } from './TraversableWithIndex'
+import { Unfoldable1 } from './Unfoldable'
+import { Witherable1 } from './Witherable'
+
+declare module './HKT' {
+  interface URItoKind<A> {
+    readonly ReadonlyArray: ReadonlyArray<A>
+  }
+}
+
+/**
+ * @since 2.5.0
+ */
+export const URI = 'ReadonlyArray'
+
+/**
+ * @since 2.5.0
+ */
+export type URI = typeof URI
+
+/**
+ * @since 2.5.0
+ */
+// tslint:disable-next-line: readonly-array
+export function fromArray<A>(as: Array<A>): ReadonlyArray<A> {
+  const l = as.length
```

Comment:
why not `identity`?

Reply:
No, I mean implementation:
```typescript
export const fromArray: <A>(as: Array<A>) => ReadonlyArray<A> = identity
```
Even more, we can just drop this function because we can safely pass `Array` as argument to functions accepting `ReadonlyArray`

Reply:
@raveclassic, with such implementation readonly array could be mutated after initialization (like in @gcanti's example above). Readonly and mutable structures should not share the same references.

Reply:
ahhh indeed